fashion-messaging-app-ui/src/App.jsx

```diff
@@ -28,42 +25,58 @@ export default function App() {
   };
   const [photos, setPhotos] = useState([]);
   const [searchQuery, setSearchQuery] = useState("");
+  const [noResults, setNoResults] = useState(false);
+
+useEffect(() => {
+  setLoading(true);
+
+  const query = searchQuery || defaultQuery;
+  clientAPI.photos
+    .search({ query, per_page: PHOTOS })
+    .then(async response => { 
+      setLoading(false);
+
+      var myHeaders = new Headers();
+      myHeaders.append("x-api-key", "1a4092c4563cc701627b1616fa6053190c6b538fd82b828d3fe20416bcc3dd3e");
+
+      let fetchPromises = await Promise.all(response.photos.map(async photo => { 
+        var formdata = new FormData();
+        formdata.append("image_url", photo.src.original);
+
+        var requestOptions = {
+          method: 'POST',
+          headers: myHeaders,
+          body: formdata,
+        };
+
+        try {
+          const fetchResponse = await fetch('https://cloudapi.lykdat.com/v1/detection/items', requestOptions);
+          const result = await fetchResponse.json();
+
+          if (result.data.detected_items.length === 0) {
+          } else {
+            result.data.detected_items.forEach(item => {
+            });
+
+            return photo;
+          }
+        } catch (error) {
+          console.error('error', error);
+        }
+      }));
+
+      const fashionPhotos = fetchPromises.filter(photo => photo !== undefined);
+      setPhotos(fashionPhotos);
+      setNoResults(fashionPhotos.length === 0);
+    })
+    .catch(error => {
+      console.error("Error fetching photos:", error);
+      setLoading(false);
+    });
+
+  localStorage.setItem('user', JSON.stringify(user));
+}, [searchQuery]);
```

Comment:
> What does this do?
- line 72-73 is error handling
- line 74 updates my loading state
-line 77 I am storing the current user in my local storage and this helps during login, so that if the user refreshes the page, they stay logged in
-line 78  Since useEfffect is trigged each time the searchQuery changes I passed searchQuery as a dependency to ensure this action takes place